src/Service/UpdateVideoItem/UpdateVideoItemRequest.php

```diff
@@ -0,0 +1,83 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace Drupal\media_mpx\Service\UpdateVideoItem;
+
+use Drupal\media\Entity\Media;
+use GuzzleHttp\Psr7\Uri;
+
+/**
+ * Contains the data needed to update a media entity via UpdateVideoItem.
+ *
+ * @package Drupal\media_mpx\Service\UpdateVideoItem
+ */
+class UpdateVideoItemRequest {
+
+  const MPX_MEDIA_DATA_SERVICE_URL = "http://data.media.theplatform.com/media/data/Media/";
+
+  /**
+   * The mpx global item id.
+   *
+   * @var \GuzzleHttp\Psr7\Uri
+   */
+  private $mpxId;
+
+  /**
+   * The media type id (bundle).
+   *
+   * @var string
+   */
+  private $mediaTypeId;
+
+  /**
+   * UpdateVideoItemRequest constructor.
+   *
+   * @param \GuzzleHttp\Psr7\Uri $mpxId
+   *   The Uri object specifying the video item id.
+   * @param string $mediaTypeId
+   *   The media type id.
+   */
+  private function __construct(Uri $mpxId, string $mediaTypeId) {
+    $this->mpxId = $mpxId;
+    $this->mediaTypeId = $mediaTypeId;
+  }
```

Comment:
constructor is private so that it's clear callers need to use one of the existing factory methods. For now there's just one, but as I'm refactoring the Importer, that'll need its own constructor, as it provides the data directly with the mpx id and the media type id.

Reply:
_what_ you can make private constructors!!??! 💥

Reply:
php is the new java 😆